### PR TITLE
initialize paging in the loader and not longer in the kernel

### PR DESCRIPTION
**Pull request by stlankes**
To be more flexible, the configuration of the processor is moved from libhermit to the loader. In addition,  paging is enabled. Consequently, the boot process is similar to x86.

### CODE REVIEW

File: src/arch/aarch64/bootinfo.rs [hunk 1]
```diff
@@ -7,11 +7,13 @@ pub struct BootInfo {
 	pub magic_number: u32,
 	pub version: u32,
 	pub base: u64,
+	pub ram_start: u64,
 	pub limit: u64,
 	pub image_size: u64,
 	pub tls_start: u64,
 	pub tls_filesz: u64,
 	pub tls_memsz: u64,
+	pub tls_align: u64,
```
Comment (mkroening) [inline src/arch/aarch64/bootinfo.rs:16]: Would we want to keep `BootInfo`'s structure as similar as possible across the architectures? For x86, I added it to the end to disrupt as little as possible. If we shuffle around the positions for x86 as well, we must pay attention to succeed the CI dance.

**Review by mkroening — COMMENTED** [review]
One slight note regarding `BootInfo`'s structure. I also moved forward with the upgrade of the toolchain separately in https://github.com/hermitcore/rusty-loader/pull/53, so this should be rebased after https://github.com/hermitcore/rusty-loader/pull/53 has been merged.

Apart from that, this looks good to me though. :+1:

File: src/arch/aarch64/bootinfo.rs (hunk 1 above)
Comment (stlankes) [reply src/arch/aarch64/bootinfo.rs:16]: Do we need that both architectures have the BootInfo? Inlong-term the arch64 port should uses device trees. I am not sure if we do it also on x86.

Comment (mkroening) [reply src/arch/aarch64/bootinfo.rs:16]: Ah, okay. If we plan to move away from `BootInfo` on aarch64, it's not as important to keep them synchronized. :+1:

**Review by mkroening — APPROVED** [review]
bors r+

**Pull request merged**